advanced/firmware/src/bin/usb-4-solution.rs

```diff
@@ -87,7 +88,9 @@ fn ep0setup(usbd: &USBD, ep0in: &mut Ep0In, state: &mut State) -> Result<(), ()>
 
     let request = Request::parse(bmrequesttype, brequest, wvalue, windex, wlength)
         .expect("Error parsing request");
-    log::info!("EP0: {:?}", request);
+    defmt::info!("EP0: {:?}", defmt::Debug2Format(&request));
```

Comment:
feel free to send a PR to [`usb2`](https://github.com/japaric/usb2) to add `derive(Format)` to these structs. I can quickly merge that and put out a new crates.io release